sections/testingandquality/aaa.md

```diff
@@ -0,0 +1,61 @@
+# Structure tests by the AAA pattern
+
+<br/><br/>
+
+### One Paragraph Explainer
+Our biggest testing challenge is the lack of headspace - we already have the production code keeping us super-busy. For this reason the testing code must stay dead-simple and easy to understand. When reading a test case - it shouldn't feel like reading imperative code (loops, inheritance) rather more like HTML - a declarative experience. To achieve this, keep the AAA convention so the readers mind will parse the test intent effortlessly. There some other similar formats to this pattern, like XUnit 'Setup, Excercise, Verify, Teardown'. These are the three A:
+
+The 1st A - Arrange: All the setup code to bring the system to the scenario the test aims to simulate. This might include instantiating the unit under test constructor, adding DB records, mocking/stubbing on objects and any other preparation code

```

Comment:
Maybe we could use headings for improved layouts or mark the titles „The 1st A...“ as bold 😄

Reply:
@BrunoScheufler Can you suggest/show something concurrent?